notebooks/concepts/specviz_glue_unit_conversion.ipynb

```diff
@@ -0,0 +1,421 @@
+    "y axis label is Data values\n",
+    "x label is Frequency \\[Hz\\]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a6aa2289-d6f0-4f0e-969a-0e559f74d00a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "\n",
+    "import numpy as np\n",
+    "from astropy.wcs import WCS\n",
+    "import glue_jupyter as gj\n",
+    "from glue.core import Data\n",
+    "from glue.plugins.wcs_autolinking.wcs_autolinking import WCSLink\n",
+    "from specutils import Spectrum1D\n",
+    "from glue.core.link_helpers import LinkSame\n",
+    "from glue.core.roi import XRangeROI\n",
+    "from glue.config import settings\n",
+    "\n",
+    "\n",
+    "from jdaviz import Specviz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "541bf389-6198-4d07-b744-4164d2ecfec4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def units_to_strings(unit_list):\n",
+    "    \"\"\"Convert equivalencies into readable versions of the units.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    unit_list : list\n",
+    "        List of either `astropy.units` or strings that can be converted\n",
+    "        to `astropy.units`.\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    result : list\n",
+    "        A list of the units with their best (i.e., most readable) string version.\n",
+    "    \"\"\"\n",
+    "    all_units = []\n",
+    "    for unit in unit_list: \n",
+    "        try:\n",
+    "            u.Unit(unit)\n",
+    "        except TypeError as te:\n",
+    "            # print(f\"{unit}: {te}\")\n",
+    "            # warnings.warn(f\"{unit}: {te}\")\n",
+    "            continue\n",
+    "        if u.Unit(unit) == u.Unit(\"Angstrom\"):\n",
+    "            all_units.append(u.Unit(unit.name))\n",
+    "        elif hasattr(u.Unit(unit), \"long_names\") and len(u.Unit(unit).long_names) > 0:\n",
+    "            all_units.append(u.Unit(unit).long_names[0])\n",
+    "        else:\n",
+    "            all_units.append(u.Unit(unit).to_string())\n",
+    "            \n",
+    "    return all_units\n",
+    "        \n",
+    "\n",
+    "    # return [u.Unit(unit).name\n",
+    "    #         if u.Unit(unit) == u.Unit(\"Angstrom\")\n",
+    "    #         else u.Unit(unit).long_names[0] if (\n",
+    "    #             hasattr(u.Unit(unit), \"long_names\") and len(u.Unit(unit).long_names) > 0)\n",
+    "    #         else u.Unit(unit).to_string()\n",
+    "    #         for unit in unit_list]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2e098d56-90d7-4b34-839f-bcd86f1bf5be",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from astropy import units as u\n",
+    "from glue.core.units import unit_converter\n",
+    "\n",
+    "@unit_converter('custom-jdaviz')\n",
+    "class UnitConverterWithSpectral:\n",
```

Comment:
I feel like this is overly complicated but I also don't know how glue set up the unit support. The tests at https://github.com/glue-viz/glue/blob/main/glue/core/tests/test_units.py are minimal and does not tell me much when it comes to equivalencies.

In principle, `astropy.units` should do the conversion for you out of the box. All we need to do is detect if it is spectral axis or flux:

* spectral axis: use `u.spectral()`
* flux: use `u.spectral_densitry(spectral_axis)`

There should be no need to hardcode so many things. The Unit Conversion plugin already should limit what the users can choose from, so probably okay to assume the input is well-behaved.

I might be missing something here though. Please help me understand. Thanks!

Reply:
I spent about an hour trying to move the code to use something like this and with those changes I was unable to get the data loading into the spectrum viewer at all! I'm definitely still in the learning stages as well with glue unit conversion so it might just be my lack of familiarity. I would welcome you trying to get it working if you can!

Reply:
Okay, lemme see. Maybe I will convince my Past Self that I was wrong lol.

Reply:
@javerbukh , does https://github.com/javerbukh/jdaviz/pull/14 work as intended or did I miss the mark? 😬